### PR TITLE
Estimate Cost command added 

### PR DESCRIPTION
**Pull request by Soumyaraj-Biswal**

### CODE REVIEW

File: extension-client/src/command/workspace/estimateCost.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import EstimateCostView from '../../webview/workspace/EstimateCostView';
+
+ 
+export async function cost(context: vscode.ExtensionContext): Promise<void> {
+    try {
+            const writeEmitter = new vscode.EventEmitter<string>();
+            const closeEmitter = new vscode.EventEmitter<number>();
+            const pty = {
+                onDidWrite: writeEmitter.event,
+                onDidClose: closeEmitter.event,
```
Comment (VishwaKumar) [inline extension-client/src/command/workspace/estimateCost.ts]: We have a terminal util [here](https://github.com/IBM-Cloud/vscode-ibmcloud-schematics/blob/main/extension-client/src/util/terminal.ts). Any particular reason this was not used here ?

File: tutorial/README.md [hunk 2]
```diff
@@ -200,3 +200,14 @@ You can also open the latest job/activity log by:
...
+
+**How to perform cost estimation?**
+
+1. Download [tfcost Binary](https://github.com/IBM-Cloud/terraform-cost-estimator/releases) & put the binary in your PATH
+2. Select View > Command Palette... from VS Code menu bar ( Keyboard Shortcut: ⇧⌘P )
+3. Select IBM Cloud Schematics workspace: Estimate Cost
+4. Select target API to deploy. Here select cloud.ibm.com from the picker
+5. Enter API key. Type in your cloud.ibm.com API key and hit enter
+6. A new WebView with estimated cost of all resources is displayed
+
```
Comment (VishwaKumar) [inline tutorial/README.md]: These steps won't be sufficient for a non-terraform users who is using this tutorial first time for cost estimation. We need to provide a example repo in the steps which will help users to try this out. Specifically a step between 1 and 2 that says to clone a repo url and then open it in vscode and then run the cost estimation command

Comment (VishwaKumar) [reply tutorial/README.md]: Also the screenshot which is added below needs to be changed to a screenshot showing the file explorer with the example template from the previous comment.

File: extension-client/src/api/index.ts [hunk 3]
```diff
@@ -20,6 +20,11 @@ const schematicsV1 = require('@ibm-cloud/ibm-schematics/dist/schematics/v1');
 import * as auth from '../auth/auth';
 import * as type from '../type/index';
 import * as util from '../util';
+import * as shell from '../command/shell';
+import { posix } from 'path';
```
Comment (VishwaKumar) [inline extension-client/src/api/index.ts]: `posix` import is not used anywhere in the code

File: extension-client/src/api/index.ts [hunk 4]
```diff
@@ -20,6 +20,11 @@ const schematicsV1 = require('@ibm-cloud/ibm-schematics/dist/schematics/v1');
...
+ import * as vscode from 'vscode';
```
Comment (VishwaKumar) [inline extension-client/src/api/index.ts]: This needs to be at the top of the other imports

File: extension-client/src/api/index.ts [hunk 5]
```diff
@@ -242,8 +247,9 @@ export async function pullLatest(data: any) {
 }
 
 export async function versions() {
+    
     const schematicsService = await auth.getSchematicsService();
-
+
```
Comment (VishwaKumar) [inline extension-client/src/api/index.ts]: remove unnecessary new lines

File: extension-client/src/command/shell/terraform/index.ts [hunk 6]
```diff
@@ -58,3 +57,60 @@ export async function hcltojsonFunc() {
...
+    const TERRAFORM_COST_COMMAND = `tfcost plan .vscode-ibmcloud-schematics/${folderName}.json --json`;
+    try{
+        await util.workspace.createCredentialFile();
+        writeEmitter.fire('\x1b[1m\x1b[33m' + "Running terraform init" + '\x1b[0m' + newLine);
+        await init();
+        writeEmitter.fire('\x1b[1m\x1b[32mSuccess!\x1b[0m ' + "terraform init" + newLine + extraNewline);
+        writeEmitter.fire('\x1b[1m\x1b[33m' + "Running terraform plan" + '\x1b[0m' + newLine);
+        await shell.execute(TERRAFORM_PLAN_COMMAND);
+        writeEmitter.fire('\x1b[1m\x1b[32mSuccess!\x1b[0m ' + "terraform plan" + newLine + extraNewline);
+        writeEmitter.fire('\x1b[1m\x1b[33m' + "Creating cost.json file" + '\x1b[0m' + newLine);
```
Comment (VishwaKumar) [inline extension-client/src/command/shell/terraform/index.ts]: We have a terminal util [here](https://github.com/IBM-Cloud/vscode-ibmcloud-schematics/blob/main/extension-client/src/util/terminal.ts). Any particular reason this was not used here ?

File: extension-ui/src/components/workspace/estimateCost/index.js [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+                        <StructuredListCell head>  
+                            <Tooltip
+                                direction="bottom"
+                                triggerText="Total estimated cost"
+                            >
+                                <p style={{padding:'2px',fontSize:"15px" ,width:'170px',background: '#0f62fe'}}>
+                                    The cost displayed here are just an estimated cost not an actual cost.
+                                    * indicated cost has been derived from ratecard
+                                </p>
+                            </Tooltip>
```
Comment (VishwaKumar) [inline extension-ui/src/components/workspace/estimateCost/index.js]: image

- The tooltip is not showing as tooltip and displaying with blue background
- The tooltip info icon needs to be beside the **Total estimated cost** and not below  
- If we cannot make the tooltip work, just have a helper text below the **Total estimated cost**

**Review by VishwaKumar — CHANGES_REQUESTED** [review]

File: tutorial/README.md (hunk 2 above)
Comment (Soumyaraj-Biswal) [reply tutorial/README.md]: We have added the additional line in the tutorial to try out the command with a template after clonning
also we updated the screenshot accordingly

File: extension-client/src/api/index.ts (hunk 3 above)
Comment (Soumyaraj-Biswal) [reply extension-client/src/api/index.ts]: Those imports have now been removed

File: extension-client/src/api/index.ts (hunk 4 above)
Comment (Soumyaraj-Biswal) [reply extension-client/src/api/index.ts]: We have removed the imports as they were not in use

File: extension-client/src/api/index.ts (hunk 5 above)
Comment (Soumyaraj-Biswal) [reply extension-client/src/api/index.ts]: We formatted the code and removed extra spaces

File: extension-ui/src/components/workspace/estimateCost/index.js (hunk 7 above)
Comment (Soumyaraj-Biswal) [reply extension-ui/src/components/workspace/estimateCost/index.js]: Tooltip was causing some design issues so we removed it and added a helper text in footer of the webview

File: extension-client/src/command/workspace/estimateCost.ts (hunk 1 above)
Comment (shipra101) [reply extension-client/src/command/workspace/estimateCost.ts]: made changes & have used terminal util

**Review by VishwaKumar — APPROVED** [review]

**Pull request merged**